dbms/src/IO/ReadWriteBufferFromHTTP.cpp

```diff
@@ -1 +1,40 @@
 #include <IO/ReadWriteBufferFromHTTP.h>
+#include <Common/Exception.h>
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+	extern const int TOO_MANY_REDIRECTS; 
+}
+
+
+std::unique_ptr<DB::ReadWriteBufferFromHTTP> makeReadWriteBufferFromHTTP(const Poco::URI & uri,
+	const std::string & method,
+	std::function<void(std::ostream &)> callback,
+	const DB::ConnectionTimeouts & timeouts,
+	const DB::SettingUInt64 max_redirects)
+	{
+		auto actual_uri =uri;
+		UInt64 redirects = 0;
+		
+		do 
+		{
+			try 
+			{
+				return std::make_unique<DB::ReadWriteBufferFromHTTP>(actual_uri, method, callback, timeouts);
+			} 
+			catch (Poco::URIRedirection & exc) {
+				redirects++;
```

Comment:
try/catch is not a good way to implement application logic.

Reply:
I agree of course. I considered that the other approach I took into account (include the number of redirects in the constructor, change receiveResponse and implement the redirection loop in the ReadWriteBufferFromHTTPBase constructor) had more risk (many more changes) and that the functionality implemented was not worth the change having a much simpler solution (without risk) although not as elegant.